### PR TITLE
[WIP] Queue Breadcrumbs Suggestion

### PR DESCRIPTION
**Pull request by nickheiner-usds**
In response to @amprokop asking for suggestions on #4565. I don't necessarily advocate merging this as-is.

With #4565's approach, we redirect the user to `/reader`, and then make Reader customizable to make it feel like it's still part of Queue. (This requirement adds considerable complexity. Hearings Prep doesn't do this. Do we really need it for Queue? Does it really provide value to end-users to think about different Caseflow apps the way we do? Should we just have one app name and logo color and just stop worrying about all this?)

The approach I explored here was embedding Reader within Queue. Passing props felt more natural than setting querystring params that Reader would have to know how to interpret, or adding one-off customizations to Reader to change things like the header app name.

This is not without downsides. We have to remember to pass `<Reader />` the same components it would get from `react_component`. And the "back to links" and breadcrumbs are not totally working and felt fairly complicated to set up. And I still am doing some querystring magic. (The need for this would be removed if react-router gave us access to every path we've visited in the SPA, but I don't think it does. We could look into adding something like this ourselves.)

### CODE REVIEW

File: app/views/queue/index.html.erb [hunk 1]
```diff
@@ -4,6 +4,12 @@
     userId: current_user.id,
     dropdownUrls: dropdown_urls,
     feedbackUrl: feedback_url,
-    buildDate: build_date
+    buildDate: build_date,
+
+    # Passthrough for Reader. This needs to be kept in sync with the react_component() call for Reader.
```
Comment (nickheiner-usds) [inline app/views/queue/index.html.erb]: We could probably find an idiomatic Rails way to share this but I didn't want to go into that now.

File: client/app/queue/QueueApp.jsx [hunk 2]
```diff
@@ -34,23 +36,52 @@ const searchStyling = (isRequestingAppealsUsingVeteranId) => css({
   }
 });
 
+const basename = '/queue';
+
+const segmentStyling = css({
+  marginTop: '3rem'
+});
+
+class BackToQueueLink extends React.PureComponent {
```
Comment (nickheiner-usds) [inline client/app/queue/QueueApp.jsx]: We would put this component in its own file; I just put it here for the rough draft.

File: client/app/queue/QueueApp.jsx [hunk 3]
```diff
@@ -34,23 +36,52 @@ const searchStyling = (isRequestingAppealsUsingVeteranId) => css({
...
+  render() {
+    const href = decodeURI(querystring.parse(this.props.location.search.slice(1)).backToPathname) || '/';
+    let text = 'Back to Your Queue';
+
+    if (href !== '/') {
```
Comment (nickheiner-usds) [inline client/app/queue/QueueApp.jsx]: I'm not crazy about how this logic works.

**Comment by amprokop** [comment]
@NickHeiner — thanks for this PR! Seems like this is a viable approach, although it does add some complexity, as you mention. After a bunch of back and forth with design, I think we settled on an approach that doesn't require this. For now, since the querystring approach to passing info is easiest and we have a launch in 3 days, IMO we should stick with that. I'm down to reevaluate how we pass info back and forth from Reader to Queue though.

**Pull request closed without merging**